apps/angular/src/app/features/auth/login/login.component.ts

```diff
@@ -0,0 +1,72 @@
+import { ChangeDetectionStrategy, Component, DestroyRef, inject } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { FormControl, FormGroup, NonNullableFormBuilder, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { AuthService } from '@js-camp/angular/core/services/auth.service';
+import { catchApiError } from '@js-camp/angular/core/utils/rxjs/catch-api-error';
+import { ApiError } from '@js-camp/core/models/api-error';
+import { AuthenticationConstants } from '@js-camp/core/utils/authentication-constants';
+import { of, tap } from 'rxjs';
+
+/** Login form controls. */
+interface LoginFormControls {
+
+	/** Email control. */
+	readonly email: FormControl<string>;
+
+	/** Password control. */
+	readonly password: FormControl<string>;
+}
+
+/** Login component. */
+@Component({
+	selector: 'login',
+	templateUrl: './login.component.html',
+	styleUrls: ['./login.component.css'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LoginComponent {
+
+	private readonly authService = inject(AuthService);
+
+	private readonly router = inject(Router);
+
+	private readonly formBuilder = inject(NonNullableFormBuilder);
+
+	private readonly destroyRef = inject(DestroyRef);
+
+	/** Login form group. */
+	protected readonly formGroup: FormGroup<LoginFormControls>;
+
+	public constructor() {
+		this.formGroup = this.formBuilder.group({
+			email: ['', [Validators.required, Validators.email]],
+			password: ['', [Validators.required, Validators.minLength(AuthenticationConstants.minPasswordLength)]],
+		});
+	}
+
+	/** Handle login form submitting. */
+	protected handleSubmit(): void {
+		if (this.formGroup.invalid) {
+			return;
+		}
+
+		const formData = this.formGroup.getRawValue();
+
+		this.authService.login({
+			email: formData.email,
+			password: formData.password,
+		}).pipe(
+			tap(() => this.router.navigate([''])),
+			catchApiError(apiError => of(this.catchLoginError(apiError))),
+			takeUntilDestroyed(this.destroyRef),
+		)
+			.subscribe();
+	}
+
+	private catchLoginError(apiError: ApiError): void {
+		this.formGroup.setErrors({
+			login: apiError.errorMessages.join(' '),
+		});
+	}
```

Comment:
As far as I can see it will be more convenient if we just return an Observable here. So we can omit 'of' operator in 'catchApiError'.
RxJS `EMPTY` can help us. It can be used here like this:
```suggestion
	private catchLoginError(apiError: ApiError): Observable<void> {
		this.formGroup.setErrors({
			login: apiError.errorMessages.join(' '),
		});
		return EMPTY;
	}
```

`EMPTY` will complete the stream. It's expected behavior when we have an error.
You can read more about it [here](https://www.learnrxjs.io/learn-rxjs/operators/creation/empty).